components/admin/index/list.module.tsx

```diff
@@ -0,0 +1,106 @@
+import styles from "./list.module.scss";
+import Link from "next/link";
+import { motion } from "framer-motion";
+import { Status } from "@typeFiles/api/project.type";
+import Tag from "@typeFiles/api/tag.type";
+
+export default function AdminPageList({ entries: entries_data, path, query }: Props): JSX.Element {
+
+  // Do a full text search on dataset.
+  const filter = (value: string): Entry[] => {
+    value = value.toLowerCase();
+    if (entries_data.length === 0) return [];
+
+    const result = entries_data.filter((entry) => {
+      if (entry.name.toLowerCase().includes(value)) return true;
+      if (entry.description?.toLowerCase().includes(value)) return true;
+      if (entry.markdown?.toLowerCase().includes(value)) return true;
+      if (entry.tags?.find((x) => x.name.toLowerCase().includes(value)))
+        return true;
+    });
+
+    return result;
+  };
+
+  const animation_item = {
+    visible: { opacity: 1, x: 0 },
+    hidden: { opacity: 0, x: -10 },
+  };
+  const animation_list = {
+    visible: { opacity: 1 },
+    hidden: { opacity: 0 },
+  };
+
+  const entries = filter(query);
```

Comment:
If entries ever becomes really big, it might be worth using promise  / async logic for the filter. just a thought.